### ISSUE TITLE
`cfg!(unsafe(foo))` and `--check-cfg=unsafe(cfg(foo))` are erroneously accepted on beta

### ISSUE THREAD

**Issue by Urgau** [rust-lang/rust#131055]
`cfg!(unsafe(foo))` and `--check-cfg=unsafe(cfg(foo))` both are erroneously accepted on beta.

They should **not** be accepted, neither of them is in any way "unsafe".

### Code

I tried this code:

```rust
fn main() {
    let _a = cfg!(unsafe(foo));
}
```

I expected to see this happen: an error saying that `unsafe` cannot be put here like for `#[cfg]`&`#[cfg_attr]`.

Instead, this happened: the `unsafe` is silently accepted.

### Version it worked on

It most recently worked on: 1.81

### Version with regression

`rustc --version --verbose`:
```
rustc 1.82.0-beta.5 (6a3b69c6b 2024-09-27)
binary: rustc
commit-hash: 6a3b69c6b0529151da5fb4568961519a80adccf9
commit-date: 2024-09-27
host: x86_64-unknown-linux-gnu
release: 1.82.0-beta.5
LLVM version: 19.1.0
```

@rustbot modify labels: +regression-from-stable-to-beta -regression-untriaged

**Comment by Noratrieb** [comment]
`unsafe { 1 }` is also accepted, so I feel like it makes sense to accept it

**Comment by Urgau** [comment]
@Noratrieb except that neither `#[cfg(unsafe(foo)]` nor `#[cfg_attr(unsafe(foo), ...)]` is accepted.

```
error: expected identifier, found keyword `unsafe`
 --> src/main.rs:1:7
  |
1 | #[cfg(unsafe(foo))]
  |       ^^^^^^ expected identifier, found keyword
  |
help: escape `unsafe` to use it as an identifier
  |
1 | #[cfg(r#unsafe(foo))]
  |       ++
```

**Comment by Noratrieb** [comment]
makes sense

**Comment by Urgau** [comment]
Backported in https://github.com/rust-lang/rust/pull/131423, closing.
